GTBitsOfGood/boggers
# Tian/role filter

I have completely changed how the semester filter works where the old implementation works as the following:

Fetch all users from the server -> Combine potential semester and year lists gathered from the data and put it in the filter -> Apply the filter

While URL query method applies a filter at API while we fetch data from the server, which is not compatible with the old method. In addition to that the old implementation was constructed based on the assumption that there is only one filter, as a result, I am unable to add a new filter using the old implementation. Therefore, I have changed the semester filter into the query parameter method and applied the filter at the backend API. However, due to this change, we now do need to keep updating the list of potential semesters in .../server/utils/fields.js
### How to Test

Go to the admin dashboard after login. Select fields from the dropdown and check that the correct data is filtered and displayed when applying filter such as developer or senior dev.
